### PR TITLE
feat(lambda-promtail): add cloudfront log file ingestion support

### PR DESCRIPTION
**Pull request by CCOLLOT**
**What this PR does / why we need it**:

This PR enables ingesting logs from Cloudfront log files stored in s3 (batch). 
The current setup only supports streaming Cloudfront logs through AWS Kinesis, whereas this PR implements the same flow as for  VPC Flow logs, Load Balancer logs, and Cloudtrail logs (s3 --> SQS (optional) --> Lambda Promtail --> Loki)

**Special notes for your reviewer**:
+ The Cloudfront log file format is different from the already implemented services, meaning we had to build yet another regex. AWS never bothered making all services follow the same log file naming convention but the "good" thing is that it's now very unlikely they will change it in the future.
+ The Cloudfront file name does not have any mention of the AWS account or the time of log it contains, it means we have to infer the log type from the filename format instead of finding the exact string "cloudfront" in the filename. This is why in `getLabels`, if no `type` parameter is found in the regex, we use the key corresponding to the name of the matching parser.
+ I introduced a new `parser` struct to group together several parameters specific to a type of log (and avoid relying too much on map key string matching and / or if statements for specific use cases)
+ I've been successfully running this code in several AWS environments for days.
+ I corrected a typo from my previous PR #9497 (wrong PR number in Changelog.md)
**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

File: tools/lambda-promtail/lambda-promtail/s3.go [hunk 1]
```diff
@@ -123,17 +161,17 @@ func parseS3Log(ctx context.Context, b *batch, labels map[string]string, obj io.
...
 		}
 		if printLogLine {
 			fmt.Println(log_line)
 		}
-
-		match := timestampRegex.FindStringSubmatch(log_line)
-		timestamp := time.Now()
+    
+    timestamp := time.Now()
+		match := parser.timestampRegex.FindStringSubmatch(log_line)
```
Comment (jeschkies) [inline tools/lambda-promtail/lambda-promtail/s3.go:172]: The formatting seems a little off.

**Review by jeschkies — APPROVED** [review]
Nice work. Could you quickly check the formatting?

File: tools/lambda-promtail/lambda-promtail/s3.go (hunk 1 above)
Comment (CCOLLOT) [reply tools/lambda-promtail/lambda-promtail/s3.go]: Thanks! Sure, there you go 👍

**Comment by CCOLLOT** [comment]
> Nice work. Could you quickly check the formatting?

Hi @jeschkies, could we merge this PR now? 🙂

**Pull request merged**